modules/electrophysiology_browser/ajax/get_electrophysiology_session_data.php

```diff
@@ -0,0 +1,336 @@
+    || ($user->hasCenter($timePoint->getCenterID())
+    && $user->hasPermission('electrophysiology_browser_view_site'));
+
+if (!$hasAccess) {
+    header('HTTP/1.1 403 Forbidden');
+    exit;
+}
+
+$response = getSessionData(intval($_REQUEST['sessionID']));
+
+echo json_encode($response);
+
+/**
+ * Get the session data information.
+ *
+ * @param int $sessionID ID of the electrophysiology session
+ *
+ * @return array with the session information
+ */
+function getSessionData(int $sessionID)
+{
+    $db = \NDB_Factory::singleton()->database();
+
+    $query = 'SELECT 
+                DISTINCT(pf.SessionID) 
+              FROM physiological_file pf 
+                LEFT JOIN session s ON (s.ID=pf.SessionID) 
+                LEFT JOIN candidate c USING (CandID)
+                LEFT JOIN psc ON (s.CenterID=psc.CenterID) 
+                LEFT JOIN physiological_output_type pot 
+                  USING (PhysiologicalOutputTypeID) 
+              WHERE 
+                s.Active = "Y" 
+                AND pf.FileType IN ("bdf", "cnt", "edf", "set", "vhdr", "vsm") 
+              ORDER BY pf.SessionID';
+
+    $response = array();
+
+    $sessions            = $db->pselect($query, array());
+    $sessions            = array_column($sessions, 'SessionID');
+    $response['patient'] = getSubjectData($sessionID);
+    $response['database']    = array_values(getFilesData($sessionID));
+    $response['sessions']    = $sessions;
+    $currentIndex            = array_search($sessionID, $sessions);
+    $response['nextSession'] = $sessions[$currentIndex+1] ?? '';
+    $response['prevSession'] = $sessions[$currentIndex-1] ?? '';
+
+    return $response;
+}
+
+/**
+ * Get the subject data information.
+ *
+ * @param int $sessionID ID of the electrophysiology session
+ *
+ * @return array with the subject information
+ */
+function getSubjectData(int $sessionID)
+{
+    $subjectData = array();
+    $timePoint   = \NDB_Factory::singleton()->timepoint($sessionID);
+    $candidate   = \NDB_Factory::singleton()->candidate($timePoint->getCandID());
+
+    $subjectData['pscid']       = $candidate->getPSCID();
+    $subjectData['dccid']       = $candidate->getCandID();
+    $subjectData['visit_label'] = $timePoint->getVisitLabel();
+    $subjectData['sessionID']   = $sessionID;
+    $subjectData['site']        = $timePoint->getPSC();
+    $subjectData['dob']         = $candidate->getCandidateDoB();
+    $subjectData['sex']         = $candidate->getCandidateSex();
+    $subjectData['subproject']  = $timePoint->getData('SubprojectTitle');
+    $subjectData['output_type'] = htmlentities(
+        $_REQUEST['outputType'],
+        ENT_QUOTES,
+        "UTF-8"
+    );
+
+    return $subjectData;
+}
+
+/**
+ * Get the list of electrophysiology recordings with their recording information.
+ *
+ * @param int $sessionID ID of the electrophysiology session
+ *
+ * @return array with the file collection
+ */
+function getFilesData(int $sessionID)
```

Comment:
Again, sessionID not required. There is a timePoint already instantiated at line 26.

Reply:
it is required to have for the query. Otherwise I could pass $timePoint but then I would have to add a line
```
$sessionID = $timepoint->getSessionID;
```
in this case, I am not sure of the benefit of this. Maybe there is something I don't get. Very likely, lol

Reply:
right